mdformat/plugins.py

```diff
@@ -15,3 +20,33 @@ def _load_codeformatters() -> Dict[str, Callable[[str, str], str]]:
 
 
 CODEFORMATTERS: Mapping[str, Callable[[str, str], str]] = _load_codeformatters()
+
+
+class ParsePluginAbstract:
+    """A class for extending the base parser."""
+
+    @staticmethod
+    def update_mdit(mdit: MarkdownIt) -> MarkdownIt:
+        """Update the parser, e.g. by adding a plugin: `mdit.use(myplugin)`"""
+        return mdit
+
+    @staticmethod
+    def render_token(
+        renderer: MDRenderer, tokens: List[Token], index: int, options: dict, env: dict
+    ) -> Optional[Tuple[str, int]]:
+        """Convert a token to a string, or return None if no render method
+        available.
+
+        :returns: (text, index) where index is of the final "consumed" token
+        """
+        return None
+
+
+def _load_extendplugins() -> Mapping[str, ParsePluginAbstract]:
```

Comment:
I would make the return type `Dict` instead of `Mapping` here.

Reply:
I copied that from `CODEFORMATTERS = ...` 😉 so I guess that should also be changed?

Reply:
I mean just `_load_extendplugins` here.

`_load_codeformatters` actually is already typehinted as `-> Dict[...`, as that is the accurate type it returns.

For the public constant `CODEFORMATTERS` I then set it to `Mapping` (which is immutable), to make type checker take care that nobody mutates the dict.

Reply:
and I see, that makes sense